### PR TITLE
Add support to chained tag matching

### PR DESCRIPTION
**Pull request by tm-jdelapuente**
**TLDR**: this PR adds support for "chained tags" where the URL specifies multiple tag values for each of the 3 tags (interpreter, abi or platform).

**Context**

There are 3 compatibility tags: interpreter, abi and platform ([see the Python Packaging User Guide on this](https://packaging.python.org/en/latest/specifications/platform-compatibility-tags/#overview)).

PyPI packages specify the tags that they support. The Wheel Resolver compares these values against the tags provided by the user to the Wheel Resolver. This prevents the user from downloading packages that are incompatible with the system they're using.

**Problem**
`_is_compatible` matches a chained tag value (e.g. `py2.py3`) against the system's equivalent tag (e.g. `py3`) rather than treating it as 2 different values (i.e. `py2` and `py3`).

This prevents the Wheel Resolver from downloading wheels that use chained tag values.

**Proposed change**
Extend `_is_compatible` to create combinations of multiple potential tag values for a URL when it uses chained tag values. Each of these tag values would be compared against the system tags.

The new test shows how this works.

### CODE REVIEW

File: tools/wheel_resolver/wheel.py [hunk 1]
```diff
@@ -74,4 +75,16 @@ def _is_wheel(url: str) -> bool:
 
 
 def _is_compatible(url: str, tags: typing.List[str]) -> bool:
-    return any(t in url for t in tags)
+    url_tags = extract_wheel_tags(url)
+    return any(t in url_tags for t in tags)
+
+
+def extract_wheel_tags(url: str) -> list[str]:
+    _, _, interpreter, abi, platform = url.replace(".whl", "").split("/")[-1].split("-")
```
Comment (peterebden) [inline tools/wheel_resolver/wheel.py:83]: would `removesuffix(".whl")` be slightly more specific?

File: tools/wheel_resolver/wheel.py [hunk 2]
```diff
@@ -74,4 +75,16 @@ def _is_wheel(url: str) -> bool:
...
+    interpreters, abis, platforms = (
```
Comment (peterebden) [inline tools/wheel_resolver/wheel.py:84]: I feel like these would be better written just one on each line, that takes three lines instead of five

**Review by peterebden — COMMENTED** [review]
Thanks!

File: tools/wheel_resolver/wheel.py (hunk 2 above)
Comment (tm-jdelapuente) [reply tools/wheel_resolver/wheel.py]: Fair, updated

**Review by peterebden — APPROVED** [review]

File: tools/wheel_resolver/wheel.py (hunk 1 above)
Comment (tm-jdelapuente) [reply tools/wheel_resolver/wheel.py]: I didn't know that one — thanks!
